app/assets/javascripts/discourse/controllers/topic.js.es6

```diff
@@ -410,6 +410,17 @@ export default Ember.Controller.extend(SelectedPostsCount, BufferedContent, {
       });
     },
 
+    mergePosts() {
+      bootbox.confirm(I18n.t("post.merge.confirm", { count: this.get('selectedPostsCount')}), result => {
+        if (result) {
+          const selectedPosts = this.get('selectedPosts');
+
+          Discourse.Post.mergePosts(selectedPosts);
```

Comment:
Let's import the `Post` model instead.
`import Post from 'discourse/models/post';`